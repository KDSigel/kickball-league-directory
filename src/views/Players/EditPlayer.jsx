import React, { useEffect, useState } from 'react'
import { useHistory, useParams } from 'react-router-dom/cjs/react-router-dom.min'
import { getPlayerById, updatePlayerById } from '../../services/players'

export default function EditPlayer() {

    const [loading, setLoading] = useState(true)
    const {id} = useParams()
    const history = useHistory()
    const [name, setName] = useState('')
    const [position, setPosition] = useState('')

    const handleEdit = async (e) => {
        e.preventDefault()
        const response = await updatePlayerById(id, { name, position })
        history.push(`/player/${response[0].id}`)
    }

    useEffect(() => {
        async function getPlayerDetails() {
            const response = await getPlayerById(id)
            setName(response.name)
            setPosition(response.position)
            setLoading(false)
        }
        getPlayerDetails()
    }, [id]
    )

    return (
        <div>
            {loading ? <h1>'loading'</h1>
                    : <>
                <fieldset>
                    <legend>Edit Player</legend>
                    <form onSubmit={handleEdit}>

                        <label htmlFor='playerName'>Player Name</label>
                        <input 
                        aria-label='player name'
                        type='text' 
                        id='teamName' 
                        name='name'
                        value={name}
                        onChange={(e) => setName(e.target.value)}
                        />

                        <label htmlFor='position'>Position</label>
                        <input 
                        aria-label='position'
                        type='text' 
                        id='position' 
                        name='position'
                        value={position}
                        onChange={(e) => setPosition(e.target.value)}
                        />

                        <button
                        aria-label='Update Player'
                        type='submit'>Update Player
                        </button>
                    </form>
                </fieldset>
            </>
            }
        </div>
    )
}
