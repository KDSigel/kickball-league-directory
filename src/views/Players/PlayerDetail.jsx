import React, { useEffect, useState } from 'react'
import { Link, useHistory, useParams } from 'react-router-dom'
import { deletePlayerById, getPlayerById } from '../../services/players'

export default function PlayerDetail() {

    const [player, setPlayer] = useState({})
    const [loading, setLoading] = useState(true)
    const {id} = useParams()
    const history = useHistory()

useEffect(() => {
    async function getPlayerDeets() {
        const response = await getPlayerById(id)
        setPlayer(response)
        setLoading(false)
    }
    getPlayerDeets()
}, [id]
)

    const handleEdit = () => {
    history.push(`/player/edit/${id}`)
    }

    const handleDelete = async () => {
        await deletePlayerById(id);
        await history.push('/players/')
    }

    return (
    
        <div>
            {loading ? <h1>'loading'</h1>
                    : <>
                    <h2>{player.name}</h2>
                    <p>{player.position} for the {player.teams.name}</p>
                    </>
            }
                                <button 
                        type='button' 
                        onClick={handleEdit}
                        >
                            Edit player
                    </button>
                    <button
                        type="button"
                        aria-label='delete player'
                        onClick={handleDelete}
                        >
                            Delete
                    </button>
        </div>
    )
}
