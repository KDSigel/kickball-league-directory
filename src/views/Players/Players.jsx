import React from 'react'
import PlayerList from '../../components/List/PlayerList'

export default function Players() {
    return (
        <div>
            <PlayerList />
        </div>
    )
}
