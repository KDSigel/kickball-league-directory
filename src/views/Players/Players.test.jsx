import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Players from './Players';

it('it should check to make sure Players loads', async () => {

const {container} = render (
<MemoryRouter>
    <Players />
</MemoryRouter>
)

screen.getByText('loading', {exact: false})
const title = await screen.findByText('Playas')
expect(title).toBeInTheDocument()

expect(container).toMatchSnapshot()

}
)