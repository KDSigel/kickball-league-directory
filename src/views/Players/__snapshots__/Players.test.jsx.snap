// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should check to make sure Players loads 1`] = `
<div>
  <div>
    <div>
      Playas
      <ul>
        <li>
          <a
            href="/player/24"
          >
            <div>
              Chris Fontaine
            </div>
          </a>
        </li>
        <li>
          <a
            href="/player/1"
          >
            <div>
              John Appleseed
            </div>
          </a>
        </li>
      </ul>
      <div>
        <fieldset>
          <legend>
            Add New Player
          </legend>
          <form>
            <label
              for="playerName"
            >
              Player Name
            </label>
            <input
              aria-label="player name"
              id="playerName"
              name="name"
              type="text"
              value=""
            />
            <label
              for="position"
            >
              Player position
            </label>
            <input
              aria-label="position"
              id="position"
              name="position"
              type="text"
              value=""
            />
            <label
              for="teamId"
            >
              Team ID
            </label>
            <input
              aria-label="team id number"
              id="teamId"
              name="teamId"
              type="text"
              value=""
            />
            <button
              aria-label="Add New Player"
              type="submit"
            >
              Add New Player
            </button>
          </form>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
